src/js/components/main/board.js

```diff
@@ -0,0 +1,293 @@
+    'с',
+    'м',
+    'и',
+    'т',
+    'ь',
+    'б',
+    'ю',
+    '.',
+    '↑',
+    'Shift',
+    'Ctrl',
+    'Win',
+    'Alt',
+    'Space',
+    'Alt',
+    '←',
+    '↓',
+    '→',
+    'Ctrl',
+  ],
+  en: [
+    '`',
+    '1',
+    '2',
+    '3',
+    '4',
+    '5',
+    '6',
+    '7',
+    '8',
+    '9',
+    '0',
+    '-',
+    '=',
+    'Backspace',
+    'Tab',
+    'q',
+    'w',
+    'e',
+    'r',
+    't',
+    'y',
+    'u',
+    'i',
+    'o',
+    'p',
+    '[',
+    ']',
+    '\\',
+    'Del',
+    'CapsLock',
+    'a',
+    's',
+    'd',
+    'f',
+    'g',
+    'h',
+    'j',
+    'k',
+    'l',
+    ';',
+    "'",
+    'Enter',
+    'Shift',
+    'z',
+    'x',
+    'c',
+    'v',
+    'b',
+    'n',
+    'm',
+    ',',
+    '.',
+    '/',
+    '↑',
+    'Shift',
+    'Ctrl',
+    'Win',
+    'Alt',
+    'Space',
+    'Alt',
+    '←',
+    '↓',
+    '→',
+    'Ctrl',
+  ],
+  keycodeEn: [
+    '`',
+    '1',
+    '2',
+    '3',
+    '4',
+    '5',
+    '6',
+    '7',
+    '8',
+    '9',
+    '0',
+    '-',
+    '=',
+    'Backspace',
+    'Tab',
+    'q',
+    'w',
+    'e',
+    'r',
+    't',
+    'y',
+    'u',
+    'i',
+    'o',
+    'p',
+    '[',
+    ']',
+    '\\',
+    'Delete',
+    'CapsLock',
+    'a',
+    's',
+    'd',
+    'f',
+    'g',
+    'h',
+    'j',
+    'k',
+    'l',
+    ';',
+    "'",
+    'Enter',
+    'Shift',
+    'z',
+    'x',
+    'c',
+    'v',
+    'b',
+    'n',
+    'm',
+    ',',
+    '.',
+    '/',
+    'ArrowUp',
+    'Shift',
+    'Control',
+    'Meta',
+    'Alt',
+    ' ',
+    'Alt',
+    'ArrowLeft',
+    'ArrowDown',
+    'ArrowRight',
+    'Control',
+  ],
+  keycodeRu: [
+    'ё',
+    '1',
+    '2',
+    '3',
+    '4',
+    '5',
+    '6',
+    '7',
+    '8',
+    '9',
+    '0',
+    '-',
+    '=',
+    'Backspace',
+    'Tab',
+    'й',
+    'ц',
+    'у',
+    'к',
+    'е',
+    'н',
+    'г',
+    'ш',
+    'щ',
+    'з',
+    'х',
+    'ъ',
+    '\\',
+    'Delete',
+    'CapsLock',
+    'ф',
+    'ы',
+    'в',
+    'а',
+    'п',
+    'р',
+    'о',
+    'л',
+    'д',
+    'ж',
+    'э',
+    'Enter',
+    'Shift',
+    'я',
+    'ч',
+    'с',
+    'м',
+    'и',
+    'т',
+    'ь',
+    'б',
+    'ю',
+    '.',
+    'ArrowUp',
+    'Shift',
+    'Control',
+    'Meta',
+    'Alt',
+    ' ',
+    'Alt',
+    'ArrowLeft',
+    'ArrowDown',
+    'ArrowRight',
+    'Control',
+  ],
+};
+
+export function generateButtons(lang = 'en') {
+  const keymap = keyCodes[lang];
+  const keyButtonContainer = new DocumentFragment();
+  keymap.map((key, i) => {
+    const keyButton = document.createElement('button');
+    keyButton.classList.add('keyboard__button');
+    keyButton.classList.add(key.toLowerCase());
+    if (lang === 'en') {
+      keyButton.dataset.keyCode = keyCodes.keycodeEn[i];
+    } else {
+      keyButton.dataset.keyCode = keyCodes.keycodeRu[i];
+    }
+    keyButton.textContent = key;
+    keyButtonContainer.append(keyButton);
+    return true;
```

Comment:
_map calls a provided callbackFn function once for each element in an array, in order, and constructs a new array from the results._

In this case you don't need a new array, so it would be more natural to use forEach method if you're keeping the current logic. However I'd recommend to adjust it a bit according the points below:

1. having a separate function that builds key html code/element and returns it would be nice.
2. inside key creator function you can use plain HTML template, it would be more visual and easy to follow.
3. when you have such function, you can go through keyMap and add each key html code/element to your fragment.

Reply:
returned lust structure of createKeyBoard, becouse innerHTML deletes textarea